gamemodes/terrortown/gamemode/server/sv_player.lua

```diff
@@ -672,12 +672,20 @@ local deathsounds = {
 }
 local deathsounds_count = #deathsounds
 
-local function PlayDeathSound(victim)
+local function PlayDeathSound(victim, isSilent)
     if not IsValid(victim) then
         return
     end
 
-    sound.Play(deathsounds[math.random(deathsounds_count)], victim:GetShootPos(), 90, 100)
+    local tbl = {
+        victim = victim,
+        sound = deathsounds[math.random(deathsounds_count)],
+        silent = isSilent,
+    }
+
+    if not hook.Run("TTT2PlayDeathScream", tbl) or tbl["silent"] then return end
```

Comment:
pulling in `isSilent` into the function doesn't make sense in my opinion.
With this there are two ways to prevent the death scream which complicates things.

IF we go with this hook implementation (which i'm fine with) we should either handle the isSilent part as we did before (old lines 792ff.) and prevent the call to `PlayDeathSound` outright. Or we should handle the isSilent part in the gamemodehook (new lines 1662ff.) you've introduced

Reply:
I understand.
Though I'm unsure how I would get the isSilent into the hook otherwise?
For me it makes perfect sense to pull isSilent into the function.. But maybe I'm just dumb.

I understand that I should use isSilent in the gm hook I introduced, so if isSilent is true > cancel it outright if it ever goes to the gm hook and is not canceled earlier.. but I still need to pass isSilent to the hook then.

I guess I could take out the isSilent from the table and make it a direct parameter in the hook..

Reply:
You are right, without `isSilent` we won't be able to handle this in the gamemodehook

In that case i'd rather use the old `if..then` approach

That way we (as the gamemode) still determine first if a death is silent, if it is not we allow addons to hook into this and either change the deathsound or silence the death (by returning false in the hook)
so no `isSilent` would be necessary

Reply:
Well.. I would agree with you but find it a bit more complicated that way..

For example: If I would check isSilent in the gamemode and not even call the Scream Hook if it is, other addons would need to use a Death Hook to play the sound if they want to play it even if it should be silent.. Actually they would need to then hook both the Scream and the Death hook and check in the deathhook if it was silent themselves.. Which makes this all way more complicated.

If I just pass isSilent to the hook too, I would just say
`if isSilent then return false end`
in the GM hook and remove the check from the PlayDeathSound function.

Reply:
Ah i see. I was giving feedback based on what you wrote in the PR description

> This hook allows to change the sound of the deathscream or cancel it outright.

If the hook should allow addons to "unsilence" a playerdeath then that's a different situation

Reply:
> Ah i see. I was giving feedback based on what you wrote in the PR description
> 
> > This hook allows to change the sound of the deathscream or cancel it outright.
> 
> If the hook should allow addons to "unsilence" a playerdeath then that's a different situation

Welp.. Maybe my description was a bit unclear, sorry for that :')

But yes, I think it would be more beneficial if the hook did both. It would at least make it more versatile.